cdc/puller/sorter/unified_sorter.go

```diff
@@ -223,6 +224,11 @@ func (s *UnifiedSorter) Run(ctx context.Context) error {
 						default:
 						}
 						err := sorter.poolHandle.AddEvent(subctx, event)
+						if cerror.ErrWorkerPoolHandleCancelled.Equal(errors.Cause(err)) {
```

Comment:
```suggestion
						if cerror.ErrWorkerPoolHandleCancelled.Equal(err) {
```
`Equal` will unwrap the err first